sql/core/src/main/scala/org/apache/spark/sql/execution/SortExec.scala

```diff
@@ -181,4 +189,17 @@ case class SortExec(
        |$sorterVariable.insertRow((UnsafeRow)${row.value});
      """.stripMargin
   }
+
+  /**
+   * In SortExec, we overwrites cleanupResources to close UnsafeExternalRowSorter.
+   */
+  override protected[sql] def cleanupResources(): Unit = {
+    super.cleanupResources()
```

Comment:
shall we release itself before releasing the children?

Reply:
Copy, done in 631f3cb.